### PR TITLE
Do not R2R NativeAOT compiler on ARM64 machines

### PR DESCRIPTION
**Pull request by MichalStrehovsky**

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

File: src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj [hunk 1]
```diff
@@ -11,7 +11,8 @@
   <PropertyGroup Condition="'$(BuildingInsideVisualStudio)' != 'true'">
     <PublishDir>$(RuntimeBinDir)ilc-published/</PublishDir>
     <PublishTrimmed>true</PublishTrimmed>
-    <PublishReadyToRun>true</PublishReadyToRun>
+    <!-- Don't R2R on ARM64 machines because ARM64 crossgen2 that comes with .NET SDK <= 7.0 Preview 6 crashes.-->
```
Comment (AntonLapounov) [inline src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj:14]: I believe we are using Preview 7 (7.0.0-preview.7.22375.6) at the moment.

**Comment by a74nh** [comment]
This fixes my build issues. Thanks for raising this.

Am I right in thinking eventually this will be reverted once nuget gets updated?

**Comment by MichalStrehovsky** [comment]
> Am I right in thinking eventually this will be reverted once nuget gets updated?

Yep.

**Comment by AndyAyersMS** [comment]
Is it possible this changed behavior of crossgen2? We are seeing some hard to explain regressions:

https://github.com/dotnet/perf-autofiling-issues/issues/8205
https://github.com/dotnet/perf-autofiling-issues/issues/8204

**Comment by MichalStrehovsky** [comment]
> Is it possible this changed behavior of crossgen2

This doesn't affect the way crossgen2 builds. This is not even using a crossgen2 from this repo, but a crossgen2 that comes from the SDK.

**Pull request merged**